js/src/components/tree-select-control/options.js

```diff
@@ -60,16 +63,18 @@ const Options = ( {
 					className="woocommerce-tree-select-control__node"
 					justify="flex-start"
 				>
-					<Icon
-						className={ classnames(
-							'woocommerce-tree-select-control__expander-icon',
-							! hasChildren ? 'is-hidden' : false
-						) }
-						icon={ isExpanded ? chevronUp : chevronDown }
-						onClick={ () => {
-							toggleExpanded( option.value );
-						} }
-					/>
+					{ ! isRoot( option ) && (
+						<Icon
+							className={ classnames(
+								'woocommerce-tree-select-control__expander-icon',
+								! hasChildren ? 'is-hidden' : false
+							) }
+							icon={ isExpanded ? chevronUp : chevronDown }
+							onClick={ () => {
+								toggleExpanded( option.value );
+							} }
+						/>
+					) }
```

Comment:
❔ 💅 Just throwing an idea out, since this "select all" node is like a special type of node with no expand/collapse icon and has its own CSS class, I wonder if it might be better to have it as a specialized node like `<SelectAllOptionNode>`. From the naming, it is also related clearly to the `selectAllOption` prop that we discuss in https://github.com/woocommerce/google-listings-and-ads/pull/1294#issuecomment-1060726223.

Note that this idea also relates to specialized node like `<ParentNode>` / `<Node>` that I mentioned in https://github.com/woocommerce/google-listings-and-ads/pull/1243.

Just something to think about. 😄

Reply:
Yes, this thought together with thoughts regarding the un-nested organization of nodes would be approached in a refactor/performance phase. 

Since right now we are actually covering the specifications in a very nice performance I will give this less prioritization in order to approach other essential areas like Search functionality or Accessibility.